app/controllers/sites_controller.rb

```diff
@@ -1,5 +1,17 @@
 class SitesController < ApplicationController
   before_action :set_site
-  load_and_authorize_resource
+  load_and_authorize_resource instance_variable: :site, class: 'Site' # descendents still auth Site
   layout 'dashboard'
+
+  def update
```

Comment:
I'm with @cbeer. This should only be on the controller that deals with banner image.  Let's make this code easy to understand and avoid creating troublesome issue like you mention.